modules/TrackPileUpSubtractor.cc

```diff
@@ -146,21 +168,29 @@ void TrackPileUpSubtractor::Process()
     while((candidate = static_cast<Candidate *>(iterator->Next())))
     {
       particle = static_cast<Candidate *>(candidate->GetCandidates()->At(0));
-      const TLorentzVector &candidateMomentum = particle->Momentum;
+      const TLorentzVector &candidateMomentum = candidate->Momentum;
 
       eta = candidateMomentum.Eta();
       pt = candidateMomentum.Pt();
       phi = candidateMomentum.Phi();
       e = candidateMomentum.E();
 
-      z = particle->Position.Z();
+      z = candidate->Position.Z();
+      counter ++; 
+      //sum = pt*pt;
+      if(pt > 1.0)
+        sumPT2 += pt*pt;
 
       // apply pile-up subtraction
       // assume perfect pile-up subtraction for tracks outside fZVertexResolution
+      // cout << particle->IsRecoPU << " ParticleSumSquare : " << sumPT2 << " VertexResult : " << sumSquare << " Added SumSquare : " << zvtx << endl;
+      //cout << pt << "  " << candidate->IsPU << "  " <<  TMath::Abs(z - zvtx) << "   " << sumPT2 << "   " << sumPT2 << endl;
 
-      if(candidate->Charge != 0 && candidate->IsPU && TMath::Abs(z - zvtx) > fFormula->Eval(pt, eta, phi, e) * 1.0e3)
+      
+      if(particle->Charge != 0 && TMath::Abs(z - zvtx) > fFormula->Eval(pt, eta, phi, e) * 1.0e3)
```

Comment:
Shouldn't a time condition also appear here for the track to be regarded as pile-up?